src/dmd/dsymbol.d

```diff
@@ -1798,7 +1798,7 @@ extern (C++) final class WithScopeSymbol : ScopeDsymbol
         // Acts as proxy to the with class declaration
         Dsymbol s = null;
         Expression eold = null;
-        for (Expression e = withstate.exp; e != eold; e = resolveAliasThis(_scope, e))
+        for (Expression e = withstate.exp; e != eold; e = resolveAliasThis(_scope, e, true))
```

Comment:
When searching an identifier in the with statement target we should not issue any errors if the alias this is not chosen.

Reply:
```suggestion
        for (Expression e = withstate.exp; e && e != eold; e = resolveAliasThis(_scope, e, true))
```
`e` can be `null` in the new case.

Reply:
@RazvanN7 Response to this?

Reply:
@maxhaton  done